### ISSUE TITLE
What happens when you try to book a first dose slot for a 45+ individual and an 18+ individual?

### ISSUE THREAD

**Issue by great1113** [bombardier-gif/covid-vaccine-booking#392]
(Assume same dose, and location)

I am thinking
   - either only the hospital which have both 18+ and 45+ slots will be tried.
   - or the slot booking never happens

I would love to go through the code and understand what is going on, but I need to run it right now. And I don't want to be running it if there is no chance for success. (rather split them into two accounts)

**Comment by Nakul93** [comment]
For this scenario the script won’t proceed if selected beneficiaries are of different age grps.
Even the CoWIN portal doesn’t allow this, even if there is a centre with both age group slots open, you need to book them separately. This can’t be done in one go.

**Comment by Nakul93** [comment]
Yeah so I noticed that in our script, it only takes the min of the age, but doesn't really warn the user during input if they provide beneficiaries of different age group.
https://github.com/bombardier-gif/covid-vaccine-booking/blob/21d5e2abffc74520ae033a33df2bfbb1e1489380/src/utils.py#L1000-L1008

For your 2nd question, the CoWIN portal has 2 different slot options for booking, 1 for 18-44 age group and 1 for 45+ age group. 
So no technically we can't book a 45+ beneficiary on a 18-44 slot.

**Comment by Nakul93** [comment]
For your 2nd question, the CoWIN portal has 2 different slot options for booking, 1 for 18-44 age group and 1 for 45+ age group. 
So no technically we can't book a 45+ beneficiary on a 18-44 slot.

**Comment by mrms123** [comment]
This is a bug. 
When the 18+ vaccinations started, anyone could book that slot - and hence the code was written such. However, now every center returns a max_age_limit field as well - this needs to be updated.

**Comment by Nakul93** [comment]
@mrms123 couldn’t find the max_age_limit field. Could you please point out to which API resp shows this field

**Comment by mrms123** [comment]
My bad - I was referring to code from another script (non-pallupz forks), and it turns out that they are manually adding that field.

This issue has been referenced here as well
https://github.com/pallupz/covid-vaccine-booking/issues/218

The error obtained is also mentioned.

**Comment by Nakul93** [comment]
I'll label this as a bug, if you have a fix you can raise a PR. Although before selecting the beneficiaries it warns the user to select beneficiaries of same age grp.

**Comment by mjishnu** [comment]
i was able to book slot for 45+ and 18+ in same center after using the new version v2.0.14, since now slot are available in 18+ format in my region and the program only looks for 18+ for both parties i am able to book slots for both together now

**Comment by sid8840** [comment]
> i was able to book slot for 45+ and 18+ in same center after using the new version v2.0.14, since now slot are available in 18+ format in my region and the program only looks for 18+ for both parties i am able to book slots for both together now

it says rate limited by cowin
how did you do this?

**Comment by sid8840** [comment]
Rate-limited by CoWIN. Waiting for 5 seconds.
(You can reduce your refresh frequency. Please note that other devices/browsers using CoWIN/Umang/Arogya Setu also contribute to same limit.)

this happens

**Comment by Nakul93** [comment]
I think the rate limiting issue was resolved…anyhow it’s strange that ppl are still using this script to book slots. There are apps now which makes the process bit easier

**Comment by Nakul93** [comment]
You can checkout this repo: https://github.com/Nakul93/BooKie. It uses public APIs and doesn’t rate limit

**Comment by mjishnu** [comment]
> Rate-limited by CoWIN. Waiting for 5 seconds.
> (You can reduce your refresh frequency. Please note that other devices/browsers using CoWIN/Umang/Arogya Setu also contribute to same limit.)
> 
> this happens

just book  2 or 3 numbers at a time dont book for too many numbers at same time and keep the refresh frequency at 15.in case u get rate limited just use a vpn connect to india.

**Comment by mjishnu** [comment]
> You can checkout this repo: https://github.com/Nakul93/BooKie. It uses public APIs and doesn’t rate limit

isnt these both same script ? 
is there any new feature in it ?

**Comment by mjishnu** [comment]
> I think the rate limiting issue was resolved…anyhow it’s strange that ppl are still using this script to book slots. There are apps now which makes the process bit easier

its still hard to book slots in my region.
